core/src/main/java/tech/beshu/ror/requestcontext/AuditLogSerializer.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+
+package tech.beshu.ror.requestcontext;
+
+import tech.beshu.ror.AuditLogContext;
+
+import java.util.Map;
+
+public interface AuditLogSerializer {
+
+  Map<String, ?> createLoggableEntry(AuditLogContext context);
```

Comment:
As mentioned below, that method signature should be:

`Map<String, ?> createLoggableEntry(ResponseContext context);`

or maybe, even better, the interface should be generic:

`public interface AuditLogSerializer<T> {
    T createLoggableEntry(ResponseContext context);
}`

And then you can have a default implementation of AuditLogSerializer, that converts ResponseContext to AuditLogContext.